src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerProgressReporter.cs

```diff
@@ -45,36 +45,34 @@ public bool InProgress
                 }
             }
 
-            public event EventHandler Started
+            public event EventHandler<bool> ProgressChanged
             {
                 add
                 {
-                    _eventMap.AddEventHandler(nameof(Started), value);
+                    _eventMap.AddEventHandler(nameof(ProgressChanged), value);
                 }
 
                 remove
                 {
-                    _eventMap.RemoveEventHandler(nameof(Started), value);
+                    _eventMap.RemoveEventHandler(nameof(ProgressChanged), value);
                 }
             }
 
-            public event EventHandler Stopped
-            {
-                add
-                {
-                    _eventMap.AddEventHandler(nameof(Stopped), value);
-                }
-
-                remove
-                {
-                    _eventMap.RemoveEventHandler(nameof(Stopped), value);
-                }
-            }
+#if DEBUG
+            // while dogfooding, encountered a case where ref count not get to 0 once. but
+            // couldn't repro it again. adding this in case it happens again then I do have
+            // actionable data to find out what went wrong.
+            private string _lastCallStackDebug;
```

Comment:
:bulb: A much better approach would be updating `BeginAsyncOperation` to send an ETW event with a stack trace. Then you get a full log of stacks in a timeline when operations begin/end. We have `RoslynEventSource`, but there may be others as well.

Reply:
this is temporary. I don't think we need full new infrastructure for this.